### DISCUSSION TITLE
[Suggestion] Set setoption41 105 as default when wemo/hue emulation is active

### DISCUSSION THREAD

**Discussion by raphamotta** [arendst/Tasmota#11368] [Ideas]
Hello,

I have several issues with Alexa not working sometimes (related with #10430 ) and 2 actions solved my problem after several tests:

Allow IGMP snooping in my LAN
Set option "setoption41 105" for gratuitous ARP every 5 minutes
Not every router has the IGPM option, and I had this problem on my parents house too. Then, "setoption41 105" fixed the issue.

I suggest to put option "setoption41 105" as default when wemo/hue emulation is active to prevent this problem. Any problem to to it?

Thanks

**Comment by barbudor** [comment]
I would vote against enforcing a specific value of SO41 when WEMO or HUE emulation is enabled.
While I understand your experience shows that it helps, is 5 minutes the good value for everyone ?
What if someone needs 2 minutes or 8 minutes for other reasons (because he is using a different switch/router or who knows ?)

I would strongly advise that you share your experience by contributing to the documentation and adds some troubleshooting tips.

just my 2c

**Reply by raphamotta** [reply]
Is there any disadvantage to set setoption41 to whatever value by default? I mean, what's the problem can it generate?

**Reply by sfromis** [reply]
As it is a workaround for dealing with "hostile" routers, it would be principally wrong to enable it always.

**Reply by jaymemaurice** [reply]
Please don't. Don't encourage fixing some broken networking devices by shipping default broken network protocols.
I personally don't want ARP broadcasts from all of my Tasmota devices waking up all the devices on my network because someone else's junk router/switch can't handle IGMP or keep it's ARP state. If we didn't want our switches to be switches, we'd still buy hubs.

**Reply by jaymemaurice** [reply]
Perhaps a better use of flash memory would be to add warning to the UI. Maybe another feature might be to document this well.

**Reply by barbudor** [reply]
@raphamotta You are still free to compile your own customized binary by adding the below code in your `user_config_override.h`
```
#undef WIFI_ARP_INTERVAL      
#define WIFI_ARP_INTERVAL      105                 // [SetOption41] Send gratuitous ARP interval
```

And as suggested, sharing your experience in the doc is very valuable

**Comment by raphamotta** [comment]
Ok, thank you guys and don't be angry with my suggestion, I just want to help others to fix this issue as well (and my router is not bad, it's a Mi Router 3G and I'm using openwrt)

**Comment by Jason2866** [comment]
In short Setoption41 is a workaround for routers / APs which are not working correctly (ARP treatment)
It will be off by default, now and in future.
Btw. i have a MI router 4A-G with OpenWRT and there is no need to use SO41.
![Screenshot_20210317-214322](https://user-images.githubusercontent.com/24528715/111536095-1c89ad00-876a-11eb-98b5-886f9565454e.jpg)

So maybe checking for a newer version will solve.

AND THX for the suggestion, this discussion can be found via search engines and will help others!

**Comment by raphamotta** [comment]
Well... I suggested Gratuitous ARP every 5 minutes n the past and @Jason2866 you said "It will be off by default, now and in future.". Now, in the 10.0.0.3 update Gratuitous ARP is enabled and set to 60 seconds. I think this needs to be in discussion again @barbudor  @jaymemaurice

**Reply by sfromis** [reply]
Relevant commit: https://github.com/arendst/Tasmota/commit/1e326460bde34fe432517a5dcaf4e05fe562a4da

**Reply by Jason2866** [reply]
@raphamotta Since many Consumer AP / Routers have no ARP treatment (missing=faulty) we decided to enable by default to workaround this bug. This has nothing to do with Hue Emulation.

**Reply by raphamotta** [reply]
> @raphamotta Since many Consumer AP / Routers have no ARP treatment (missing=faulty) we decided to enable by default to workaround this bug. This has nothing to do with Hue Emulation.

I understand... but I fixed my bug updating openwrt. In the end, gratuitous ARP did not help.

**Reply by Jason2866** [reply]
As i assumed your issue had probably nothing to do with the reason why we enabled Gratuitous ARP now.
Hue packets are UDP traffic.
